core/src/main/scala/com/yahoo/maha/core/query/OffHeapRowList.scala

```diff
@@ -0,0 +1,194 @@
+package com.yahoo.maha.core.query
+
+import java.io.File
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicLong
+
+import com.yahoo.maha.core.query.QueryRowList.ROW_COUNT_ALIAS
+import com.yahoo.maha.core.request.Parameter.RequestId
+import com.yahoo.maha.rocksdb.{RocksDBAccessor, RocksDBAccessorBuilder}
+import com.yahoo.maha.serde.LongSerDe
+import grizzled.slf4j.Logging
+import org.apache.commons.io.{FileUtils, FilenameUtils}
+
+import scala.collection.mutable.ArrayBuffer
+import scala.util.Try
+
+/*
+    Created by pranavbhole on 2/3/20
+*/
+case class OffHeapRowListConfig(tempStoragePathOption:Option[String], inMemRowCountThreshold: Int = 10000)
+
+case class OffHeapRowList(query: Query, config: OffHeapRowListConfig) extends QueryRowList with AutoCloseable with Logging {
+
+  private[this] val tempStoragePath = OffHeapRowList.getAndValidateTempStorage(config.tempStoragePathOption)
+
+  private[this] var rocksDBAccessorOption: Option[RocksDBAccessor[Long, Row]] = None
```

Comment:
Storing a Row could get expensive because you're storing the alias map with each row.  In the InMem version, they all refer to 1 instance of the alias map so its not expensive, but serializing it would mean you store the copy of it with each row.  You may want to only store the Array buffer in the row.

Reply:
Created RowValue case class which only stores ArrayBuffer